### DISCUSSION TITLE
Make addSelect() not overrule the default select() value

### DISCUSSION THREAD

**Discussion by pimski** [laravel/framework#39557] [Ideas]
Hi!

When using eloquent's query builder I often rely on the default select value.
```php
User::query()->get(); // select * from users;
```

However, sometimes I need to append an extra column so I use addSelect (usually hidden in a scope on the model). This results in the  default select being overruled by the addSelect column.
```php
User::query()->someScopeWithAddSelect()->get(); // select colum_from_add_select from users;
```

It would be nice if addSelect doesn't overrule the default select value. Also, I think it's safer to default to the models table.
```php
User::query()->someScopeWithAddSelect()->get(); // select users.*, colum_from_add_select from users;
```

In summary:
- When no select is specified default to \<table\>.*
- addSelect doesn't overrule the default select value

**Comment by satalways** [comment]
I have same issue...

**Comment by HugoHeneault** [comment]
Same here, have you find a way to do it?
